pkg/auth/handler/api/workflow_new.go

```diff
@@ -145,6 +146,7 @@ func (h *WorkflowNewHandler) makeSessionOptions(cookie *http.Cookie) (*workflow.
 		RedirectURI:              uiInfo.RedirectURI,
 		SuppressIDPSessionCookie: uiInfo.SuppressIDPSessionCookie,
 		State:                    uiInfo.State,
+		XState:                   req.XState(),
```

Comment:
Add XState to UIInfo and read uiInfo.XState to set this field, just like what we did for State.